src/frontend/src/views/Upload/components/Metadata/components/StaticTable/index.tsx

```diff
@@ -50,78 +62,132 @@ const validationSchema = object({
 export default function StaticTable({
   metadata,
   setIsValid,
+  hasImportedMetadataFile,
 }: Props): JSX.Element {
+  const [validationErrors, setValidationErrors] =
+    useState<Record<string, ValidationErrorRecord | null>>(EMPTY_OBJECT);
+
+  // This function validates metadata by creating a mapping of sample ids to
+  // either null or a ValidationErrorRecord, an object which is itself a mapping
+  // from the key(s) that failed to validate to the specific error message.
+  // The function then determines if the metadata as a whole is valid by
+  // checking that each sampleId maps to null.
+  // If one field has multiple error messages we just overwrite the message in
+  // the ValidationErrorRecord, since we only want to display one error at a time.
   const validateMetadata = useCallback(
     async (metadata: SampleIdToMetadata | null) => {
       if (metadata == null) {
         setIsValid(false);
         return;
       }
-      const rowValidation: Record<string, boolean> = {};
+      const validationErrors: Record<string, ValidationErrorRecord | null> =
+        Object.fromEntries(
+          Object.keys(metadata).map((sampleId) => [sampleId, null])
+        );
       for (const [sampleId, sampleMetadata] of Object.entries(metadata)) {
-        const isRowValid = await validationSchema.isValid(sampleMetadata);
-        rowValidation[sampleId] = isRowValid;
+        try {
+          await validationSchema.validate(sampleMetadata, {
+            abortEarly: false,
+          });
+        } catch (error) {
+          if (error instanceof ValidationError) {
+            const errorRecord: ValidationErrorRecord = {};
+            error.inner.forEach((validationError) => {
+              if (validationError.path != undefined) {
+                const rootMetadataKey = validationError.path.split(".")[0];
+                errorRecord[rootMetadataKey as keyof ValidationErrorRecord] =
+                  validationError.message;
+              }
+            });
+            validationErrors[sampleId] = errorRecord;
+          } else {
+            throw error;
+          }
+        }
       }
       const isValid = Object.keys(metadata).every(
-        (sampleId) => rowValidation[sampleId]
+        (sampleId) => validationErrors[sampleId] == null
       );
+      setValidationErrors(validationErrors);
       setIsValid(isValid);
     },
-    []
+    [metadata]
   );
 
   useEffect(() => {
-    validateMetadata(metadata);
-  }, [metadata]);
+    if (hasImportedMetadataFile) {
+      validateMetadata(metadata);
+    }
+  }, [metadata, hasImportedMetadataFile]);
+
+  // Sort entries by error status, then by sampleId
+  let errorSortedMetadata: [string, Metadata][] = [];
+  if (metadata != null) {
```

Comment:
Are you intending to do a non-strict comparison here? If not, consider using `!==` (that's what we use in most of the rest of the code base)
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Equality